docs/content/docs/test/TestCaseInCode.md

```diff
@@ -3,19 +3,75 @@ title: Test case in code
 type: docs
 ---
 
+{{<avito page>}}
+
 # Test case in code
 
-Правда о тест кейсе исторически хранилась в TMS.\
-Поддерживать актуальной информацию сразу в двух местах не самое достойное занятие, а увеличение кол-ва автоматизированных
-кейсов остро поставило вопрос о стоимости этого занятия.
+Avito uses in-house TMS: [Internal docs](http://links.k.avito.ru/h)
+
+TMS was a source of truth about test cases in avito since the very early days.\
+While company shifted to test automation, manual synchronization between test cases in TMS and code in autotests became a problem.\
+
+To simplify this process "test cases in code" concept was born.\
+Now only code contains truth about test cases, and automated ones are read-only in TMS.
+
+## How to synchronize your test with TMS?
+
+Test should have this minimal set of annotations:
+ - `@ExternalId(UUID)` UUID - random generated uuid (use any tool, for example online generator)
+ - `@Description` - testcase title
+ - `@FeatureId(IntArray)` - specify test's place in global features tree
+ - `@TagId(IntArray)` - specify id's of team tag cloud
+ - [Kind]({{< ref "#kind" >}})
+ 
+-- deprecated --
+ - `@CaseId(Int)` - TMS id, consider using [test case in code]({{< ref "/docs/test/TestCaseInCode.md#test-case-in-code" >}})
```

Comment:
"/docs/test/TestCaseInCode.md#test-case-in-code" - is it a valid link?

Reply:
yes, but i think could be shorten to `#test-case-in-code` or `#`

Reply:
changed to `#test-case-in-code`